include/__threading_support

```diff
@@ -93,8 +96,14 @@ typedef void* __libcpp_recursive_mutex_t[5];
 #endif
 
 // Condition Variable
+#if defined(NXDK)
+// Size of CONDITION_VARIABLE is different in NXDK
+typedef void* __libcpp_condvar_t[4];
+#define _LIBCPP_CONDVAR_INITIALIZER {static_cast<void*>(0), static_cast<void*>(0), static_cast<void*>(0), static_cast<void*>(0)}
```

Comment:
This should probably mimick https://github.com/XboxDev/nxdk/blob/a00406664e67bb962da8269e80507e8eb7cf89e3/lib/winapi/winbase.h#L56, so it has to be `static_cast<void*>(0), static_cast<void*>(-1), static_cast<void*>(-1), static_cast<void*>(0)`.

That probably causes https://github.com/XboxDev/nxdk-libcxx/pull/2#issuecomment-665255191.

this also starts showing why using this weird `void*[4]` hack is bad. Because not all members are zero anymore, we are now dealing with signed vs. unsigned and more precise field sizes.

Reply:
I do agree that looks like a suitable fix, but doing like so gives me this compilation error:
```
In file included from /home/dracc/Projects/nxdk/lib/libcxx/include/mutex:190:
/home/dracc/Projects/nxdk/lib/libcxx/include/__mutex_base:287:32: error: cannot cast from type 'int' to pointer type 'void *'
    __libcpp_condvar_t __cv_ = _LIBCPP_CONDVAR_INITIALIZER;
                               ^~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/dracc/Projects/nxdk/lib/libcxx/include/__threading_support:102:61: note: expanded from macro '_LIBCPP_CONDVAR_INITIALIZER'
#define _LIBCPP_CONDVAR_INITIALIZER {static_cast<void*>(0), static_cast<void*>(-1), static_cast<void*>(-1), static_cast<void*>(0)}
                                                            ^~~~~~~~~~~~~~~~~~~~~~
```

Reply:
I fixed the initialization values and changed the type of the array to `int32_t` to get rid of the casts.